src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -762,17 +798,64 @@ export class SearchSource {
     return field;
   }
 
+  public async loadDataViewFields() {
+    const dataView = this.getField('index');
+    if (!dataView || !(dataView instanceof DataViewLazy)) {
+      return;
+    }
+    // eslint-disable-next-line no-console
+    console.log('loadDataViewFields because of DataViewLazy');
+    const request = this.mergeProps(this, { body: {} }, ['query', 'filter']);
+    let fields = dataView.timeFieldName ? [dataView.timeFieldName] : [];
+    const sort = this.getField('sort');
```

Comment:
Overall I find this code very useful although I need to see how it compares to the various computations in the `SearchSource.flatten` method. I think it likely covers all cases where fields need to be loaded but I need to verify.

Reply:
yes, of course this needs to be evaluated, I might have missed something, however, in local testing, adding a alert from discover with query, sorting, filters, it seemed to work (I just created the alert executing every few seconds, watching kibana log output)

Reply:
This really only concerns scripted fields, so it doesn't need to be fetched. That said, skipping the field fetch for this is an optimization that can be left for later.